src/EventStore.Core/GossipStatsCollector.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Diagnostics.Metrics;
+using EventStore.Core.Telemetry;
+
+namespace EventStore.Core;
+
+public class GossipStatsCollector {
+	private readonly DurationTracker _tracker;
+
+	internal GossipStatsCollector(DurationMetric metric, string name) {
+		_tracker = new DurationTracker(metric, name);
+	}
+
+	public IDisposable StartRequest() {
+		return _tracker.Start();
+	}
+}
+
+public class GossipStatsCollectors {
```

Comment:
to fit the pattern this class might be better of being called `GossipTrackers` (and having its own file)